bin/helpers/openstack.sh

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env bash
+
+function get_image_status() {
+  local upload_name=$1
+  local image_id
+  local image_details
+  image_id=$(get_image_id "${upload_name}")
+  image_details="$(glance image-show "${image_id}")"
+
+  printf '%s' "${image_details}" |
+    grep -w "status" |
+    awk -F "|" '{print $3;}' |
+    tr -d '[:space:]'
+}
+
+function get_image_id() {
+  local upload_name=$1
+  local image_list
+  image_list="$(glance image-list)"
+
+  printf '%s' "${image_list}" |
+    grep "${upload_name}" |
+    awk -F "|" '{print $2;}' |
+    tr -d '[:space:]'
+}
+
+function get_server_image_disc_format() {
+  local image_id=$1
+  local image_details
+  image_details="$(glance image-show "${image_id}")"
+
+  printf '%s' "${image_details}" |
+    grep -w "disk_format" |
+    awk -F "|" '{print $3;}' |
+    tr -d '[:space:]'
+}
+
+function image_exists() {
+  local upload_name=$1
+  local image_id
+  image_id=$(get_image_id "${upload_name}")
+
+  if [[ "${image_id}" != "" ]]; then
+    echo "true"
+  else
+    echo "false"
+  fi
+}
+
+function get_server_uploaded_image_checksum() {
+  local image_id=$1
+  local image_details
+  image_details="$(glance image-show "${image_id}")"
+
+  printf '%s' "${image_details}" |
+    grep -w "checksum" |
+    awk -F "|" '{print $3;}' |
+    tr -d '[:space:]'
+}
+
+function is_image_downloaded_already() {
+  local download_filepath=$1
+
+  if [[ -e ${download_filepath} ]]; then
+    echo "true"
+  else
+    echo "false"
+  fi
+}
+
+function is_uploaded_disc_format_as_expected() {
+  local image_name=$1
+  local disc_format=$2
+  local image_id
+  local upload_name
+  upload_name="${image_name}"
+  image_id=$(get_image_id "${upload_name}")
+  server_disc_format=$(get_server_image_disc_format "${image_id}")
+
+  if [[ "${server_disc_format}" == "${disc_format}" ]]; then
+    echo "true"
+  else
+    echo "false"
+  fi
+}
+
+function is_image_already_uploaded() {
+  local image_name=$1
+  local disc_format=$2
+  local image_id
+  local upload_name
+  upload_name="${image_name}"
+  image_id=$(get_image_id "${upload_name}")
+
+  # If the image exists, return true, otherwise return false
+  if [[ "${image_id}" == "" ]]; then
+    echo "false"
+  else
+    echo "true"
+  fi
+}
+
+function wait_for_image_activation() {
+  local upload_name=$1;
+  local wait_time
+  local max_wait
+  local sleep_time
+  local status
+  wait_time=0
+  max_wait=300
+  sleep_time=3
+  status="nothing"
+
+  # Wait until image status is "active"
+  while [[ "${status}" != "active" && ${wait_time} -lt ${max_wait} ]]; do
+    # allow for image to be ready
+    sleep ${sleep_time}
+    wait_time=$((wait_time + sleep_time))
+    status=$(get_image_status "${upload_name}")
+  done
+
+  echo "${status}"
+}
+
+function get_host_uploaded_image_checksum() {
+  local image_name=$1
+  local disc_format=$2
```

Comment:
To avoid confusion with image format on image downloaded change "disk_format" to "glance_disk_format"

Reply:
:+1:  - i did this for all `disc_format` - i presume that is what you mean ?